client/src/components/Map/Map.js

```diff
@@ -1,26 +1,123 @@
 import React, { Component } from 'react';
-import ReactMapGL from 'react-map-gl';
+import ReactMapGL, {Marker, Popup, GeolocateControl} from 'react-map-gl';
+// import mapboxgl from 'mapbox-gl';
+import './Map.css';
+import placeIcon from '../../assets/images/PlaceIcon.png'
 
 class Map extends Component {
-    state = {
-        viewport: {
-          width: 400,
-          height: 400,
-          latitude: 37.7577,
-          longitude: -122.4376,
-          zoom: 8
+    constructor(props) {
+        super(props);
+        this.state = {
+            viewport: {
+                width: '100vw',
+                height: '100vh',
+                latitude: 37.7577,
+                longitude: -122.4376,
+                zoom: 12
+              },
+              selectedRestaurant: null,
+              restaurants: [{
+                              type: "Restuarant",
+                              features: {
+                                  Id: 960,
+                                  Name: "Nob Hill Cafe",
+                                  Reviews: 5,
+                                  Address: "1152 Taylor St, San Francisco, CA 94108",
+                                  Hours: null,
+                                  Description: "This relaxed cafe serves Tuscan pastas & pizzas in a quaint space that recalls old San Francisco",
+                                  FoodType: ["Comfort Food", "Pizza"],
+                                  Allergies: ["Peanuts", "Lentils"]
+                              },
+                              geometry: {
+                                  Coordinates: [37.792980, -122.435790]
+                              }
+                          }
+              ]
         }
+        this.setSelectedRestaurant = this.setSelectedRestaurant.bind(this);
+        this.closeEffect = this.closeEffect.bind(this);
+        this.locateUser = this.locateUser.bind(this);
     };
+    componentDidMount(){
+        
+    }
+    locateUser() {
+        // https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/Using_geolocation
+        navigator.geolocation.getCurrentPosition(async(position) => {
+          await this.setState({viewport:{
+            longitude: position.coords.longitude,
+            latitude: position.coords.latitude,
+            zoom: 8
+          }});
+        });
+      }
+    // getRestaurant(){
+    //     getRestaurant()
+    //     //will need an api call to get all the restaurants within the viewport 
+    //     //can then take this data and render on the map with the Marker component
+    //     //alternatively, we could preload the data for faster load times 
+    // }
+    async setSelectedRestaurant(event, restaurant){
+        event.preventDefault();
+        await this.setState({selectedRestaurant:restaurant})
+    }
+    closeEffect(){
```

Comment:
if you make it an arrow function, you don't need to bind on line 38.
ex)
closeEffect = () => {
  //you can use "this" within the scope of this class
}